dashboard/client/src/components/ActorTable.tsx

```diff
@@ -99,11 +91,63 @@ const ActorTable = ({
   const [actorIdFilterValue, setActorIdFilterValue] = useState(filterToActorId);
   const [pageSize, setPageSize] = useState(10);
 
+  const uptimeSorterKey = "fake_uptime_attr";
+  const gpuUtilizationSorterKey = "fake_gpu_attr";
+  const gramUsageSorterKey = "fake_gram_attr";
+
+  const defaultSorterKey = uptimeSorterKey;
+  const { sorterFunc, setOrderDesc, setSortKey, sorterKey, descVal } =
+    useSorter(defaultSorterKey);
+
   //We get a filtered and sorted actor list to render from prop actors
   const sortedActors = useMemo(() => {
+    const aggregateUserSortKeys = [
+      uptimeSorterKey,
+      gpuUtilizationSorterKey,
+      gramUsageSorterKey,
+    ];
     const actorList = Object.values(actors || {}).filter(filterFunc);
-    return sortActors(actorList);
-  }, [actors, filterFunc]);
+    let actorsSortedUserKey = actorList;
+    if (aggregateUserSortKeys.includes(sorterKey)) {
+      // Uptime, GPU utilization, and GRAM usage are user specified sort keys but require an aggregate function
+      // over the actor attribute, so sorting with sortBy
+      actorsSortedUserKey = _.sortBy(actorList, (actor) => {
+        const descMultiplier = descVal ? 1 : -1;
+        switch (sorterKey) {
+          case uptimeSorterKey:
+            const startTime = actor.startTime;
+            const endTime = actor.endTime;
+            // If actor doesn't have startTime, set uptime to infinity for sort so it appears at the bottom of
+            // the table by default
+            const uptime =
+              startTime && startTime > 0
+                ? getDurationVal({ startTime, endTime })
```

Comment:
I just realized because duration is very dynamic field (changes every second), the sorting of this list may not update when the durations change.

I think this is probably fine... an edge case most people may not notice. Can you add a comment stating we will only re-sort rows on re-renders.